packages/shared/src/hooks/post/usePostImage.ts

```diff
@@ -0,0 +1,18 @@
+import { useMemo } from 'react';
+import { Post } from '../../graphql/posts';
+
+export const usePostImage = (post: Post): string =>
+  useMemo(() => {
+    if (post?.image) {
+      return post?.image;
+    }
+
+    const parser = new DOMParser();
+    const doc = parser.parseFromString(post?.contentHtml, 'text/html');
+    const imgTag = doc.querySelector('img');
+    if (imgTag) {
+      return imgTag.getAttribute('src');
+    }
+
+    return undefined;
```

Comment:
Do we want to return undefined or null?

Reply:
I think undefined should be alright. Though I just copy-pasted this from the original implementation.